packages/@webex/media-helpers/test/unit/spec/webrtc-core.js

```diff
@@ -1,45 +1,130 @@
-        })
+      afterEach(() => {
+        sinon.restore();
+      });
 
-        it('rejects setMute(false) if unmute is not allowed', async () => {
-          track.setUnmuteAllowed(false);
+      it('by default allows unmuting', async () => {
+        assert.equal(track.isUnmuteAllowed(), true);
+        await track.setMuted(false);
+      })
 
-          assert.equal(track.isUnmuteAllowed(), false);
-          await assert.isRejected(track.setMuted(false), Error, 'Unmute is not allowed');
-        });
+      it('rejects setMute(false) if unmute is not allowed', async () => {
+        track.setUnmuteAllowed(false);
 
-        it('resolves setMute(false) if unmute is allowed', async () => {
-          track.setUnmuteAllowed(true);
+        assert.equal(track.isUnmuteAllowed(), false);
+        const fn = () => track.setMuted(false);
+        expect(fn).to.throw(/Unmute is not allowed/);
+      });
 
-          assert.equal(track.isUnmuteAllowed(), true);
-          await track.setMuted(false);
-        })
+      it('resolves setMute(false) if unmute is allowed', async () => {
+        track.setUnmuteAllowed(true);
+
+        assert.equal(track.isUnmuteAllowed(), true);
+        track.setMuted(false);
+      });
+
+      it('tests setServerMuted- true to false', async () => {
+        track.underlyingTrack.enabled = true; //start with unmuted
+
+        track.setMuted(true); //this.muted is true, call setServerMuted with false
+        assert.equal(track.muted, true);
+
+        const spy = sinon.stub(track, 'emit');
+        track.setServerMuted(false, 'remotelyMuted');
+
+        assert.equal(track.muted, false);
+
+        const spyCall = spy.getCall(1);
+        assert.equal(spyCall.calledWithExactly(event.ServerMuted, {muted: false, reason: 'remotelyMuted' }), true);
+      });
+
+      it('tests setServerMuted- false to true', async () => {
+        track.underlyingTrack.enabled = false; //start with muted
+
+        track.setMuted(false); //this.muted is false, call setServerMuted with true
+        assert.equal(track.muted, false);
+
+        const spy = sinon.stub(track, 'emit');
+        track.setServerMuted(true, 'remotelyMuted');
+
+        assert.equal(track.muted, true);
+
+        const spyCall = spy.getCall(1);
+        assert.equal(spyCall.calledWithExactly(event.ServerMuted, {muted: true, reason: 'remotelyMuted' }), true);
+      });
+
+      it('tests setServerMuted - true to true', async () => {
+        track.underlyingTrack.enabled = true; //start with unmuted
+
+        track.setMuted(true); //this.muted is true, call setServerMuted with true
+        assert.equal(track.muted, true);
+
+        const spy = sinon.stub(track, 'emit');
+        track.setServerMuted(true, 'remotelyMuted');
+
+        assert.equal(track.muted, true);
+
+        const spyCall = spy.getCall(1);
+        assert.equal(spyCall, null);
+      });
+
+      it('tests setServerMuted- false to false', async () => {
```

Comment:
this test and 3 previous ones are almost exactly the same, they could be refactored to re-use code from same helper test function